androlo/bytes/ByteSlice.sol

```diff
@@ -0,0 +1,171 @@
+/// Used to create slices from 'bytes'.
+contract ByteSlice {
+
+    struct Slice {
+        uint _unsafe_memPtr;   // Memory address of the first byte.
+        uint _unsafe_len;      // Length.
+    }
+
+    /// Create a slice from bytes.
+    function fromBytes(bytes memory bts) internal constant returns (Slice memory slice) {
+        uint memPtr;
+        assembly {
+            memPtr := add(bts, 0x20)
+        }
+        slice = Slice(memPtr, bts.length);
+    }
+
+    /// Length of the slice.
+    function len(Slice memory slice) internal constant returns (uint) {
+        return slice._unsafe_len;
+    }
+
+    /// returns the byte at 'index'.
+    /// Index must be less then 'len(slice)', or the function will throw.
+    function at(Slice memory slice, uint index) internal constant returns (byte b) {
+        if (index >= slice._unsafe_len)
+            throw;
+        uint start = slice._unsafe_memPtr;
+        uint bb;
+        assembly {
+            bb := byte(0, mload(add(start, index)))
+        }
+        b = byte(bb);
```

Comment:
It's an internal method so can't do that, but I have updated it to read into `b` directly.